src/main/java/io/lakefs/routerfs/PathMapper.java

```diff
@@ -177,13 +197,19 @@ private MappingConfig parseMappingConf(Map.Entry<String, String> hadoopConf) thr
      * @param origPath the path to map
      * @return a mapped path
      */
-    public Path mapPath(Path origPath) {
-        PathMapping pathMapping = findAppropriatePathMapping(origPath);
+    public PathProperties mapPath(Path origPath) {
+        PathMapping pathMapping = findAppropriatePathMapping(origPath)
+                .orElseGet(() -> findDefaultPathMapping(origPath).orElse(null));
```

Comment:
Chains of orElses are easier to read when not nested, I think.  Can we do this?
```suggestion
        PathMapping pathMapping = findAppropriatePathMapping(origPath)
                .orElseGet(() -> findDefaultPathMapping(origPath))
                .orElse(null));
```

Reply:
I know it's kinda ugly but unfortunately `orElseGet` and `orElse` return a `PathMapping` and not `Optional<PathMapping>` so the nested calls are necessary...
Is it triggering a [callback hell](https://miro.medium.com/max/1400/1*zxx4iQAG4HilOIQqDKpxJw.jpeg) flashback?